File: Sqrt.java
public class Sqrt {

    static final int ERROR = -13;

    public static double[] mid_sqrt(double NUM) {
	double low = 0;
	double high = NUM;
	double ans = (low + high) / 2.0;
	double guess = ans * ans;
	int guesses = 1;
	while(Math.abs(guess - NUM) > Math.pow(10, ERROR)) {
	    if (guess < NUM) {
		low = (low + ans) / 2.0;
	    }
	    else {
		high = (high + ans) / 2.0;
	    }
	    ans = (low + high) / 2.0;
	    guess = ans * ans;
	    guesses++;
	}
	return new double[] {ans, (double) guesses};
    }

    public static double[] quot_sqrt(double NUM) {
	double ans = 1.0;
	double quotient = NUM / ans;
	int guesses = 1;
	while(Math.abs(ans * ans - NUM) > Math.pow(10, ERROR)) {
	    ans = (ans + quotient) / 2.0;
	    quotient = NUM / ans;
	    guesses++;
	}
	return new double[] {ans, (double) guesses};
    }

    public static void mid_sqrt_print(double NUM) {
	double [] returned = mid_sqrt(NUM);
	System.out.printf("~ Midpoint Square Root Method ~%n");
	System.out.printf("\tThe square root of %f is %." + (-1 * ERROR - 1) + "f. (%d guesses)%n", NUM, returned[0], (int) returned[1]);
    }

    public static void quot_sqrt_print(double NUM) {
	double [] returned = quot_sqrt(NUM);
	System.out.printf("~ Quotient Square Root Method ~%n");
	System.out.printf("\tThe square root of %f is %." + (-1 * ERROR - 1) + "f. (%d guesses)%n", NUM, returned[0], (int) returned[1]);
    }

    public static void pow_sqrt_print(double NUM) {
	double ans = Math.pow(NUM, 0.5);
	System.out.printf("~ 0.5 Exponent Square Root Method ~%n");
	System.out.printf("\tThe square root of %f is %f.%n", NUM, ans);
    }

    public static void def_sqrt_print(double NUM) {
	double ans = Math.sqrt(NUM);
	System.out.printf("~ Default Square Root Method ~%n");
	System.out.printf("\tThe square root of %f is %f.%n", NUM, ans);
    }

    public static void sqrt(double NUM) {
	System.out.printf("===== %f =====%n", NUM);
	mid_sqrt_print(NUM);
	quot_sqrt_print(NUM);
	pow_sqrt_print(NUM);
	def_sqrt_print(NUM);
	System.out.println();
    }

    public static void main(String[] args) {
	System.out.println();
	for(int c = 0; c < args.length; c++) {
	    try {
		double NUM = Double.parseDouble(args[c]);
		sqrt(NUM);
	    }
	    catch (NumberFormatException ex) {
		System.out.printf("%s is not a number%n%n", args[c]);
	    }
	}
    }

}

File: sqrt.py
ERROR = -13

def mid_sqrt(NUM):
    low = 0
    hi = NUM
    ans = (low + hi) / 2.0
    guess = ans * ans
    guesses = 1
    while abs(guess - NUM) > 10 ** ERROR:
        if guess < NUM:
            low = (low + ans) / 2.0
        else:
            hi = (hi + ans) / 2.0
        ans = (low + hi) / 2.0
        guess = ans * ans
        guesses += 1
    return ans, guesses

def quot_sqrt(NUM):
    guess = 1
    quotient = float(NUM) / guess
    guesses = 1
    while abs(guess * guess - NUM) > 10 ** ERROR:
        guess = (guess + quotient) / 2.0
        quotient = float(NUM) / guess
        guesses += 1
    return guess, guesses

def mid_sqrt_print(NUM):
    ans, guesses = mid_sqrt(NUM)
    ans = round(ans, -ERROR - 1)
    print("~ Midpoint Square Root Method ~")
    print("\tSquare root of " + str(NUM) + " is " + str(ans) + ". (" + str(guesses) + " guesses)")

def quot_sqrt_print(NUM):
    ans, guesses = quot_sqrt(NUM)
    ans = round(ans, -ERROR - 1)
    print("~ Quotient Square Root Method ~")
    print("\tSquare root of " + str(NUM) + " is " + str(ans) + ". (" + str(guesses) + " guesses)")

def def_sqrt_print(NUM):
    print("~ Default Square Root Method ~")
    print("\tSquare root of " + str(NUM) + " is " + str(NUM ** 0.5) + ".")

def sqrt(NUM):
    print("===== " + str(NUM) + " =====")
    mid_sqrt_print(NUM)
    quot_sqrt_print(NUM)
    def_sqrt_print(NUM)
    print("\n")
    
def main():
    import sys
    print()
    for i in range(len(sys.argv) - 1):
        try:
            NUM = float(sys.argv[i + 1])
            sqrt(NUM)
        except ValueError:
            print(sys.argv[i + 1] + " is not a number.\n")

if __name__ == "__main__":
    main()
